Breeding-Insight/bi-api
# [BI-1831] Add DatasetID to ObservationUnits

# Description
**Story:** [BI-1831](https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?selectedIssue=BI-1831)
The Experiment processor was changed so that new observation units created during experiment import have a dataset external ref assigned with the experiment observation dataset id set as the reference id.

A previous TODO was fulfilled by updating the BrAPITrialService::getDatasetData to use only OUs that have the corresponding dataset id as an exref.

A flyway migration was created to add dataset xrefs to existing observation units belonging to BI experiemnts.

NOTE: further bug fixes in breedbase are required in order to verify the migration occurred without error, so the verification is left commented out unilt that TODO is done in a separate card: [BI-1898](https://breedinginsight.atlassian.net/browse/BI-1898)

# Dependencies
A bug in the ObservationUnits.pm update method was fixed and is needed for the migration to work with Breedbase.
Run the SGN branch `feature/BI-1831`.

# Testing
1. create two test programs: connect one to the brapi-java-server, the other to breedbase
2. switch to the `develop` branch for bi-api 
3. for both programs, import an experiment with observation variable headers but no observations
4. switch to the `feature/BI-1831` branch
5. verify that the observation units created in step 3 now have the dataset xref
6. import the same experiment with a new experiment name
7. verify that the observation units created during the import in step 6 have dataset xrefs assigned
8. download the experiment
9. verify fidelity between the export and import data


# Checklist:



[BI-1831]: https://breedinginsight.atlassian.net/browse/BI-1831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[BI-1898]: https://breedinginsight.atlassian.net/browse/BI-1898?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
